subprojects/core/src/main/java/org/gradle/execution/commandline/CommandLineTaskConfigurer.java

```diff
@@ -57,7 +60,7 @@ private List<String> configureTasksNow(Collection<Task> tasks, List<String> argu
         List<String> remainingArguments = null;
         for (Task task : tasks) {
             CommandLineParser parser = new CommandLineParser();
-            Map<Boolean, List<OptionDescriptor>> allCommandLineOptions = optionReader.getOptions(task, false).stream().collect(Collectors.groupingBy(OptionDescriptor::isClashing));
+            Map<Boolean, List<OptionDescriptor>> allCommandLineOptions = TaskOptionSupplier.get(task, optionReader, false).stream().collect(Collectors.groupingBy(OptionDescriptor::isClashing));
             List<OptionDescriptor> validCommandLineOptions = allCommandLineOptions.getOrDefault(false, Collections.emptyList());
             List<OptionDescriptor> clashingOptions = allCommandLineOptions.getOrDefault(true, Collections.emptyList());
             clashingOptions.forEach(it -> LOGGER.warn("Built-in option '{}' in task {} was disabled for clashing with another option of same name", it.getName(), task.getPath()));
```

Comment:
:art: This should be moved to `TaskOptionSupplier` so it is close to the similar logic for clashing opposite options.

Then we don't need non valid options in `CommandLineTaskConfigurer` this will simplify things.

Moreoever, the `help` task now warns about clashing opposites but not about clashing built-ins. Fixing this would make this uniform.